### DISCUSSION TITLE
Editing or Deleting one row- Impacts entire table

### DISCUSSION THREAD

**Discussion by hodbby** [dbeaver/dbeaver#35216] [Q&A] [answered]
### Description

[Screencast from 19-07-24 13:45:47.webm](https://github.com/user-attachments/assets/c3637eb2-25a9-4f1e-aafa-6fa1bb04588f)

### DBeaver Version

24.1.2.202407071610

### Operating System

Ubunto 22

### Database and driver

1. Database- PostgresSQL
2. Driver- PostgreSQL JDBC Driver
3. 

### Steps to reproduce

1. Open DBeaver on any table
2. Make sure you have a few rows
3. Modify one of the rows
4. Save
5. Now refresh table
6. **Expected**- Only that specific row to be modified
7. **Bug**- Entire table changed
8. S**ame when deleting one row- Entire rows are deleted**
9. Video attached

**Comment by ask9** [comment, marked as answer]
It is not a bug.
Add unique key to the table so that the application can identify the row you're editing.
First of all, application should warn you about it with this message and you could choose from options.
Seem like you pressed "Use all columns" here some time before:

![изображение](https://github.com/user-attachments/assets/4c4bb641-eb69-4ce6-948b-cfebebbe6b02)
